Panic when deployment has more than one container and the container is not specified in the manifest

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x2026c9f]
goroutine 35 [running]:
github.com/okteto/okteto/pkg/k8s/deployments.TranslateDevContainer(0x0, 0xc0016e4d00)
    /home/circleci/project/pkg/k8s/deployments/translate.go:180 +0x6f
github.com/okteto/okteto/pkg/k8s/deployments.translate(0xc00009e840, 0xc0015862c0, 0xc0003e2420, 0x0, 0x0)
    /home/circleci/project/pkg/k8s/deployments/translate.go:102 +0x65e
github.com/okteto/okteto/pkg/k8s/deployments.TranslateDevMode(0xc001583d40, 0xc0015862c0, 0xc0003e2420, 0xc001583d40, 0x0)
    /home/circleci/project/pkg/k8s/deployments/crud.go:180 +0xb8
github.com/okteto/okteto/cmd.(*UpContext).devMode(0xc00151e090, 0xc0015c8900, 0x0, 0x0, 0x0)
    /home/circleci/project/cmd/up.go:408 +0x3f4
github.com/okteto/okteto/cmd.(*UpContext).Activate(0xc00151e090, 0x0)
    /home/circleci/project/cmd/up.go:248 +0x353
created by github.com/okteto/okteto/cmd.RunUp
    /home/circleci/project/cmd/up.go:167 +0x1cb
```

--- a/pkg/k8s/deployments/crud.go
+++ b/pkg/k8s/deployments/crud.go
@@ -214,6 +214,10 @@ func DevModeOff(tr *model.Translation, imageToRedeploy string, c *kubernetes.Cli
 	if imageToRedeploy != "" {
 		for _, rule := range tr.Rules {
 			devContainer := GetDevContainer(&d.Spec.Template.Spec, rule.Container)
+			if devContainer == nil {
+				return fmt.Errorf("Container '%s' not found in deployment '%s'", rule.Container, t.Deployment.Name)
+			}
+
 			devContainer.Image = imageToRedeploy
 		}
 	}

--- a/pkg/k8s/deployments/crud.go
+++ b/pkg/k8s/deployments/crud.go
@@ -215,7 +215,7 @@ func DevModeOff(tr *model.Translation, imageToRedeploy string, c *kubernetes.Cli
 		for _, rule := range tr.Rules {
 			devContainer := GetDevContainer(&d.Spec.Template.Spec, rule.Container)
 			if devContainer == nil {
-				return fmt.Errorf("Container '%s' not found in deployment '%s'", rule.Container, t.Deployment.Name)
+				return fmt.Errorf("Container '%s' not found in deployment '%s'", rule.Container, d.GetName())
 			}
 
 			devContainer.Image = imageToRedeploy

--- a/pkg/k8s/deployments/translate.go
+++ b/pkg/k8s/deployments/translate.go
@@ -99,6 +99,10 @@ func translate(t *model.Translation, ns *apiv1.Namespace, c *kubernetes.Clientse
 	TranslatePodAffinity(&t.Deployment.Spec.Template.Spec, t.Name)
 	for _, rule := range t.Rules {
 		devContainer := GetDevContainer(&t.Deployment.Spec.Template.Spec, rule.Container)
+		if devContainer == nil {
+			return fmt.Errorf("Container '%s' not found in deployment '%s'", rule.Container, t.Deployment.Name)
+		}
+
 		TranslateDevContainer(devContainer, rule)
 		TranslateOktetoVolumes(&t.Deployment.Spec.Template.Spec, rule)
 		TranslatePodSecurityContext(&t.Deployment.Spec.Template.Spec, rule.SecurityContext)

